### PR TITLE
vscode: update to 1.29.1

### PR DESCRIPTION
**Pull request by shizonic**
vscode: switch package maintainer

### CODE REVIEW

**Comment by marmeladema** [comment]
Also, adding "enableTelemetry": false to product.json does not change anything for me, telemetry is still enable by default.

**Comment by shizonic** [comment]
Thanks. Will have a look into by tomorrow. But one thing to say: I do not have a files directory, only patches. The files directory was used by the last maintainer.

**Comment by marmeladema** [comment]
Yes but a patch that is here to only create a file is pretty much the same thing :)

**Comment by shizonic** [comment]
So any idea to disable telemetry?

**Comment by shizonic** [comment]
About which patch are you talking about?

**Comment by marmeladema** [comment]
I am trying other things without luck for now.

**Comment by shizonic** [comment]
Okay. Let me know if you found a solution. And thanks for your suggestions.

**Comment by marmeladema** [comment]
I was talking about srcpkgs/vscode/patches/desktop.patch but i read it too fast sorry. I still think using vscode build system to generate the proper desktop patch is better though

**Comment by shizonic** [comment]
Of course it is. I will integrate your patch by tomorrow if it's okay for you and update the PR.

**Comment by marmeladema** [comment]
Thanks ;) I also think the open file descriptor limit is going to be hit on the build server like here: https://build.voidlinux.eu/builders/i686_builder/builds/10486/steps/shell_3/logs/stdio

**Comment by shizonic** [comment]
I know. This happens sometimes and sometimes not. On my local machine I increased the limit, but I am not sure if I can do it on the build servers too.

**Comment by shizonic** [comment]
So have to sleep now. Will look into it asap.

**Comment by marmeladema** [comment]
@maxice8 do you know if the max open file descriptor limit can be increase for packaging vscode ?

**Comment by shizonic** [comment]
@marmeladema I updated the package wirh your patch.

**Comment by maxice8** [comment]
Is the old maintainer active ? and if yes did you get permission to switch ?

**Comment by shizonic** [comment]
@maxice8 The old maintainer isn't that active and did allowed me to switch. Have a look at this discussion to verify:
https://github.com/void-linux/void-packages/pull/1957#issuecomment-416892217

File: srcpkgs/vscode/template [hunk 1]
```diff
@@ -1,38 +1,57 @@
...
+_mem_limit="--max_old_space_size=4096"
+
 case "$XBPS_TARGET_MACHINE" in
 	i686)	broken="https://build.voidlinux.eu/builders/i686_builder/builds/10486/steps/shell_3/logs/stdio"
 		_ARCH="ia32";;
 	x86_64) _ARCH="x64";;
 esac
 
+post_extract() {
+	if [ "${_software_rendering}" = "yes" ]; then
```
Comment (maxice8) [inline srcpkgs/vscode/template]: I don't see a point in doing this `if then fi` control structure.

just keep the sed which is always "true" and when you don't need it anymore just comment it out/remove it

File: srcpkgs/vscode/template [hunk 2]
```diff
@@ -1,38 +1,57 @@
...
+checksum=1147b8f665cc8d025d09a5250c284d5a7a2d0d1a6bc5a97eaf3a33864b21339e
+patch_args="-Np1"
+
 # Due to electron
 only_for_archs="i686 x86_64"
 nostrip_files="code-oss"
 
+# Due to this bug:
+# https://github.com/Microsoft/vscode/issues/23946
+_software_rendering="yes"
```
Comment (maxice8) [inline srcpkgs/vscode/template]: i'd remove this and move the comment to above the sed call, also add another line with a short explaination

**Comment by shizonic** [comment]
@maxice8 Okay I fixed the template according to you suggestions.

**Comment by shizonic** [comment]
I just replaced the complete `sed` line with an additional patch.

**Comment by whoamin** [comment]
Why do you trying turn off telemetry? This is done specifically to improve the performance and usability of vscode based on anonymous statistical data. It's not for hacking you or relevant advertising.

Also, it's already 1.27.1. U must subscribe on releases feed or drop maintaining https://github.com/Microsoft/vscode/releases.atom

**Comment by maxice8** [comment]
> Why do you trying turn off telemetry? This is done specifically to improve the performance and usability of vscode based on anonymous statistical data. It's not for hacking you or relevant advertising.

Can't it be enabled later manually by the user ?

**Comment by whoamin** [comment]
It's enabled by default, and no reason to disable it. But this can always be turned off or on in the settings. 
In Arch Linux no one thought of disabling telemetry ... maybe just because it does not need to be done?

**Comment by whoamin** [comment]
https://code.visualstudio.com/docs/supporting/FAQ#_gdpr-and-vs-code

**Comment by maxice8** [comment]
> It's enabled by default, and no reason to disable it.

There is no reason to have any form of data collection on by default, if a program really wants it it can prompt the user at startup like i (believe) Ubuntu did with their installer.

> But this can always be turned off or on in the settings.

I see no problem them people that want to improve the product with their data can just flip the switch.

**Comment by maxice8** [comment]
After getting info with @Vaelatern we currently just leave telemetry option as upstream picks them.

so telemetry should be left on in this package.

**Comment by shizonic** [comment]
@whoami and @maxice Sorry for my delay. I was extremely busy the last two days. I will update the package today and enable telemetry again.

**Comment by maxice8** [comment]
You tagged the wrong person, no capital M and an 8 at the end.

**Comment by shizonic** [comment]
@maxice8 I apologize. It was right after I did get up in morning and I wrote on my mobile phone. Forgive me... :)

**Comment by shizonic** [comment]
@maxice8 Just add a patch to fix https://github.com/Microsoft/vscode/issues/58097

**Comment by shizonic** [comment]
Update to v1.27.2 will follow by tomorrow.

**Comment by shizonic** [comment]
@maxice8 Update to 1.27.2.

**Comment by whoamin** [comment]
@maxice8 reason for delay?

**Comment by the-eater** [comment]
@shizonic You may be able to cherry-pick `2ba2fa5d3ccfeb63b2f7a31050d635f6ee133ed7` from https://github.com/Microsoft/vscode/, like I had done here: https://github.com/void-linux/void-packages/pull/2904/commits/5c3bce684537118feacc84503963dae798a5dcad#diff-21160d677ae0fe91b1b31790088a6910

This would unbreak vscode :)

**Comment by shizonic** [comment]
@the-eater Nice. Thanks. Will do that asap.

**Comment by shizonic** [comment]
@maxice8 and @the-eater: Electron upgrade patch added.

**Comment by shizonic** [comment]
Updated to version 1.28.0.

File: srcpkgs/vscode/patches/libgl.patch [hunk 3]
```diff
@@ -0,0 +1,11 @@
+diff --git a/resources/linux/bin/code.sh b/resources/linux/bin/code.sh
+index 55f50b6f1c14..a083c5f8c10a 100755
+--- a/resources/linux/bin/code.sh
++++ b/resources/linux/bin/code.sh
+@@ -32,5 +32,5 @@ fi
+
+ ELECTRON="$VSCODE_PATH/@@NAME@@"
+ CLI="$VSCODE_PATH/resources/app/out/cli.js"
+-ELECTRON_RUN_AS_NODE=1 "$ELECTRON" "$CLI" "$@"
++LIBGL_ALWAYS_SOFTWARE=1 ELECTRON_RUN_AS_NODE=1 "$ELECTRON" "$CLI" "$@"
```
Comment (the-eater) [inline srcpkgs/vscode/patches/libgl.patch]: Why is this needed? :thinking:

Comment (the-eater) [reply srcpkgs/vscode/patches/libgl.patch]: Is it worth that a visual bug (that doesn't seem to get triggered often) degrades performance everywhere?

maybe something like

```sh
if [ -e "${HOME}/.config/code-oss/.libgl_always_software" ]; then
  export LIBGL_ALWAYS_SOFTWARE=1
fi
```

with a notice in the `INSTALL` would be nicer/better?

Comment (shizonic) [reply srcpkgs/vscode/patches/libgl.patch]: I am working with it since months and didn't notice any performance impact. I think gpu rendering isn't that essential for vscode. But indeed would your solution be better the problem is that if you export LIBGL_ALWAYS_SOFTWARE=1 every other application would be affected too.

Comment (the-eater) [reply srcpkgs/vscode/patches/libgl.patch]: `export` only affects everything inside the current bash script,

as long as you don't invoke this script via the `source` command in bash everything will be fine.

```sh
echo "Start: ${HELLO}";
bash -c 'export HELLO="world"; echo "Inside: ${HELLO}"';
echo "End: ${HELLO}"
```
results e.g. in:

```
Start:
Inside: world
End:
```

Comment (shizonic) [reply srcpkgs/vscode/patches/libgl.patch]: Thanks didn't know that. In this case I will update asap and implement your solution. Thanks!

Comment (the-eater) [reply srcpkgs/vscode/patches/libgl.patch]: Awesome :) thanks!

Comment (shizonic) [reply srcpkgs/vscode/patches/libgl.patch]: I have to thank you, dude. :)

**Comment by shizonic** [comment]
Update to version 1.28.1

**Comment by shizonic** [comment]
Update to version 1.28.2

**Comment by Vaelatern** [comment]
I'd be more comfortable with merging this if the patches had sources (say: the originating github repos or the like)

**Comment by shizonic** [comment]
Don't undestand. I did patch it iby myself. So I am the source...

**Comment by the-eater** [comment]
Could you add an explanation at the top of the diff files what it does and why it's included?

**Comment by shizonic** [comment]
@the-eater Of course. Will do it asap.

**Comment by Vaelatern** [comment]
And we like patches best when we are not the only place that has them.

**Comment by shizonic** [comment]
@Vaelatern I see. Please excuse me. I just wanted to get the best out of vscode.

I will remove all patches except the ones which are required to build it. Is that okay?

**Comment by Vaelatern** [comment]
Feel free to ask upstream to apply patches too!

**Comment by shizonic** [comment]
Update to version 1.29.0

**Comment by shizonic** [comment]
Update to version 1.29.1

**Comment by whoizit** [comment]
is it possible to build on musl?

**Comment by Vaelatern** [comment]
It is possible.

It's very hard.

Feel free to help fix bugs!

**Comment by shizonic** [comment]
@Vaelatern How ? First electron has to be packaged I guess?
